programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb

```diff
@@ -0,0 +1,1240 @@
+    "  facet_wrap(. ~ sequencing_platform, ncol = 1, scales = 'free') +\n",
+    "  xlab('Sample') +\n",
+    "  ylab('Heterozygosity Rate ') +\n",
+    "  ggtitle('Scatter Plot: Heterozygosity Rate on the X Chromosome')\n",
+    "if (nrow(joined_genome_results) <= 20) {\n",
+    "  p + theme(axis.text.x = element_text(angle = 50, hjust = 1))\n",
+    "} else {\n",
+    "  p + theme(axis.text.x = element_blank(), axis.ticks.x = element_blank(),\n",
+    "            panel.grid.major.x = element_blank())\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Display outliers "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(m_outliers <- joined_genome_results %>%\n",
+    "    filter(!sex %in% c('F', 'FEMALE', 'female') & perct_het_alt_in_snvs > 0.4) %>%\n",
+    "    arrange(desc(perct_het_alt_in_snvs))\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(f_outliers <- joined_genome_results %>%\n",
+    "    filter(! sex %in% c('M', 'MALE', 'male') & perct_het_alt_in_snvs < 0.4) %>%\n",
+    "    arrange(perct_het_alt_in_snvs))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "problems <- rbind(\n",
+    "    tibble(name = m_outliers$name, fail = str_glue('sample info indicates {m_outliers$sex}, genome indicates female')),\n",
+    "    tibble(name = f_outliers$name, fail = str_glue('sample info indicates {f_outliers$sex}, genome indicates male')),\n",
+    "    problems)\n",
+    "\n",
+    "dim(problems)\n",
+    "head(problems)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Problem summary\n",
+    "\n",
+    "Group all the 'problems' together by sample and emit a CSV report."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "(problem_summary <- problems %>%\n",
+    "    group_by(name) %>%\n",
+    "    summarize(\n",
+    "        issues = str_c(fail, collapse = ';\\n '),\n",
+    "        issue_count = n()\n",
+    "    ) %>%\n",
+    "    arrange(desc(issue_count), name))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "write_csv(problem_summary, file = CSV_OUTPUT_FILE_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Provenance "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "devtools::session_info()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Copyright 2022 Verily Life Sciences LLC. All rights reserved.\n",
```

Comment:
The copyright should remain the same. This is just an update to a preexisting notebook.